urlgrabber/byterange.py

```diff
@@ -17,25 +17,54 @@
 # This file is part of urlgrabber, a high-level cross-protocol url-grabber
 # Copyright 2002-2004 Michael D. Stenner, Ryan Tomayko
 
+from __future__ import print_function
+from __future__ import unicode_literals
 
 import os
 import stat
-import urllib
-import urllib2
-import rfc822
+
+try:
+    # Python2
+    from urllib import addclosehook
+    from urllib2 import FileHandler
+    from urllib2 import addinfourl
+    from urllib2 import BaseHandler
+    from urllib2 import FTPHandler
+    from urllib2 import FileHandler
+    from urllib2 import URLError
+    from urllib2 import ftpwrapper as urllib_ftpwrapper
+    from urllib2 import splitport
+except ImportError:
+    # Python3
```

Comment:
IMHO the Python3 imports should be placed first, and the deprecated names tried only as fallback.